File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/ViewFieldDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Columns for view.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=ViewFieldDetails.Builder.class)

public final class ViewFieldDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"fieldName", "fieldType", "fieldPrecision", "fieldScale", "fieldDescription"})
    public ViewFieldDetails(String fieldName, String fieldType, String fieldPrecision, String fieldScale, String fieldDescription) {
        super();
        this.fieldName = fieldName;
        this.fieldType = fieldType;
        this.fieldPrecision = fieldPrecision;
        this.fieldScale = fieldScale;
        this.fieldDescription = fieldDescription;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Name of the column.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("fieldName")
        private String fieldName;

                /**
         * Name of the column.
         * @param fieldName the value to set
         * @return this builder
         **/
        

        public Builder fieldName(String fieldName) {
        this.fieldName = fieldName;
        return this;
        }
            /**
     * Type of the column.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("fieldType")
        private String fieldType;

                /**
         * Type of the column.
         * @param fieldType the value to set
         * @return this builder
         **/
        

        public Builder fieldType(String fieldType) {
        this.fieldType = fieldType;
        return this;
        }
            /**
     * Precision of the column.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("fieldPrecision")
        private String fieldPrecision;

                /**
         * Precision of the column.
         * @param fieldPrecision the value to set
         * @return this builder
         **/
        

        public Builder fieldPrecision(String fieldPrecision) {
        this.fieldPrecision = fieldPrecision;
        return this;
        }
            /**
     * Scale of the column.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("fieldScale")
        private String fieldScale;

                /**
         * Scale of the column.
         * @param fieldScale the value to set
         * @return this builder
         **/
        

        public Builder fieldScale(String fieldScale) {
        this.fieldScale = fieldScale;
        return this;
        }
            /**
     * Description of the column.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("fieldDescription")
        private String fieldDescription;

                /**
         * Description of the column.
         * @param fieldDescription the value to set
         * @return this builder
         **/
        

        public Builder fieldDescription(String fieldDescription) {
        this.fieldDescription = fieldDescription;
        return this;
        }


        public ViewFieldDetails build() {
            ViewFieldDetails model = new ViewFieldDetails(this.fieldName
                    , this.fieldType
                    , this.fieldPrecision
                    , this.fieldScale
                    , this.fieldDescription);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(ViewFieldDetails model) {
                this.fieldName(model.getFieldName());
    this.fieldType(model.getFieldType());
    this.fieldPrecision(model.getFieldPrecision());
    this.fieldScale(model.getFieldScale());
    this.fieldDescription(model.getFieldDescription());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Name of the column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("fieldName")
    private final String fieldName;

        /**
     * Name of the column.
     * @return the value
     **/
    
    public String getFieldName() {
        return fieldName;
    }


        /**
     * Type of the column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("fieldType")
    private final String fieldType;

        /**
     * Type of the column.
     * @return the value
     **/
    
    public String getFieldType() {
        return fieldType;
    }


        /**
     * Precision of the column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("fieldPrecision")
    private final String fieldPrecision;

        /**
     * Precision of the column.
     * @return the value
     **/
    
    public String getFieldPrecision() {
        return fieldPrecision;
    }


        /**
     * Scale of the column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("fieldScale")
    private final String fieldScale;

        /**
     * Scale of the column.
     * @return the value
     **/
    
    public String getFieldScale() {
        return fieldScale;
    }


        /**
     * Description of the column.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("fieldDescription")
    private final String fieldDescription;

        /**
     * Description of the column.
     * @return the value
     **/
    
    public String getFieldDescription() {
        return fieldDescription;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("ViewFieldDetails(");
        sb.append("fieldName=").append(String.valueOf(this.fieldName));
        sb.append(", fieldType=").append(String.valueOf(this.fieldType));
        sb.append(", fieldPrecision=").append(String.valueOf(this.fieldPrecision));
        sb.append(", fieldScale=").append(String.valueOf(this.fieldScale));
        sb.append(", fieldDescription=").append(String.valueOf(this.fieldDescription));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ViewFieldDetails)) {
            return false;
        }

        ViewFieldDetails other = (ViewFieldDetails) o;
        return java.util.Objects.equals(this.fieldName, other.fieldName) &&
            java.util.Objects.equals(this.fieldType, other.fieldType) &&
            java.util.Objects.equals(this.fieldPrecision, other.fieldPrecision) &&
            java.util.Objects.equals(this.fieldScale, other.fieldScale) &&
            java.util.Objects.equals(this.fieldDescription, other.fieldDescription);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.fieldName == null ? 43 : this.fieldName.hashCode());
        result = (result * PRIME) + (this.fieldType == null ? 43 : this.fieldType.hashCode());
        result = (result * PRIME) + (this.fieldPrecision == null ? 43 : this.fieldPrecision.hashCode());
        result = (result * PRIME) + (this.fieldScale == null ? 43 : this.fieldScale.hashCode());
        result = (result * PRIME) + (this.fieldDescription == null ? 43 : this.fieldDescription.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/view_field_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class ViewFieldDetails(object):
    """
    Columns for view.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new ViewFieldDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param field_name:
            The value to assign to the field_name property of this ViewFieldDetails.
        :type field_name: str

        :param field_type:
            The value to assign to the field_type property of this ViewFieldDetails.
        :type field_type: str

        :param field_precision:
            The value to assign to the field_precision property of this ViewFieldDetails.
        :type field_precision: str

        :param field_scale:
            The value to assign to the field_scale property of this ViewFieldDetails.
        :type field_scale: str

        :param field_description:
            The value to assign to the field_description property of this ViewFieldDetails.
        :type field_description: str

        """
        self.swagger_types = {
            'field_name': 'str',
            'field_type': 'str',
            'field_precision': 'str',
            'field_scale': 'str',
            'field_description': 'str'
        }

        self.attribute_map = {
            'field_name': 'fieldName',
            'field_type': 'fieldType',
            'field_precision': 'fieldPrecision',
            'field_scale': 'fieldScale',
            'field_description': 'fieldDescription'
        }

        self._field_name = None
        self._field_type = None
        self._field_precision = None
        self._field_scale = None
        self._field_description = None

    @property
    def field_name(self):
        """
        **[Required]** Gets the field_name of this ViewFieldDetails.
        Name of the column.


        :return: The field_name of this ViewFieldDetails.
        :rtype: str
        """
        return self._field_name

    @field_name.setter
    def field_name(self, field_name):
        """
        Sets the field_name of this ViewFieldDetails.
        Name of the column.


        :param field_name: The field_name of this ViewFieldDetails.
        :type: str
        """
        self._field_name = field_name

    @property
    def field_type(self):
        """
        Gets the field_type of this ViewFieldDetails.
        Type of the column.


        :return: The field_type of this ViewFieldDetails.
        :rtype: str
        """
        return self._field_type

    @field_type.setter
    def field_type(self, field_type):
        """
        Sets the field_type of this ViewFieldDetails.
        Type of the column.


        :param field_type: The field_type of this ViewFieldDetails.
        :type: str
        """
        self._field_type = field_type

    @property
    def field_precision(self):
        """
        Gets the field_precision of this ViewFieldDetails.
        Precision of the column.


        :return: The field_precision of this ViewFieldDetails.
        :rtype: str
        """
        return self._field_precision

    @field_precision.setter
    def field_precision(self, field_precision):
        """
        Sets the field_precision of this ViewFieldDetails.
        Precision of the column.


        :param field_precision: The field_precision of this ViewFieldDetails.
        :type: str
        """
        self._field_precision = field_precision

    @property
    def field_scale(self):
        """
        Gets the field_scale of this ViewFieldDetails.
        Scale of the column.


        :return: The field_scale of this ViewFieldDetails.
        :rtype: str
        """
        return self._field_scale

    @field_scale.setter
    def field_scale(self, field_scale):
        """
        Sets the field_scale of this ViewFieldDetails.
        Scale of the column.


        :param field_scale: The field_scale of this ViewFieldDetails.
        :type: str
        """
        self._field_scale = field_scale

    @property
    def field_description(self):
        """
        Gets the field_description of this ViewFieldDetails.
        Description of the column.


        :return: The field_description of this ViewFieldDetails.
        :rtype: str
        """
        return self._field_description

    @field_description.setter
    def field_description(self, field_description):
        """
        Sets the field_description of this ViewFieldDetails.
        Description of the column.


        :param field_description: The field_description of this ViewFieldDetails.
        :type: str
        """
        self._field_description = field_description

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
